pkg/postgres/booleans.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright The CloudNativePG Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package postgres
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+)
+
+// ParsePostgresBoolean returns the boolean value parsed from a string as a postgres boolean.
+// It returns an error if the input string is not a valid postgres boolean
+// See: https://www.postgresql.org/docs/current/config-setting.html
+// Boolean: Values can be written as on, off, true, false, yes, no, 1, 0 (all case-insensitive)
+// or any unambiguous prefix of one of these.
+func ParsePostgresBoolean(in string) (bool, error) {
+	trueValues := []string{"1", "on", "yes", "true", "y", "t", "ye", "tr", "tru"}
+	falseValues := []string{"0", "no", "off", "false", "n", "f", "of", "fa", "fal", "fals"}
+
+	sanitized := strings.TrimSpace(in)
+	sanitized = strings.ToLower(sanitized)
+
+	if slices.Contains(falseValues, sanitized) {
+		return false, nil
+	}
+	if slices.Contains(trueValues, sanitized) {
+		return true, nil
+	}
+	return false, fmt.Errorf("configuration value is not a postgres boolean: %s", in)
+}
```

Comment:
What about this implementation?

```suggestion
func ParsePostgresBoolean(in string) (bool, error) {
	booleans := map[string]bool{
		"1":     true,
		"on":    true,
		"y":     true,
		"ye":    true,
		"yes":   true,
		"t":     true,
		"tr":    true,
		"tru":   true,
		"true":  true,
		"0":     false,
		"n":     false,
		"no":    false,
		"of":    false,
		"off":   false,
		"f":     false,
		"fa":    false,
		"fal":   false,
		"fals":  false,
		"false": false,
	}

	sanitized := strings.TrimSpace(in)
	sanitized = strings.ToLower(sanitized)

	if val, ok := booleans[sanitized]; ok {
		return val, nil
	}

	return false, fmt.Errorf("configuration value is not a postgres boolean: %s", in)
}
```

I find it easier to read.

Reply:
I prefer it, as you say it improve the readability.
+1

Reply:
In my opinion, we should only accept two types and not try to do more, even if Postgres accepts them. This is because, in the end, they express only two values.
I propose that we accept only "true" "false" or "on" and "off" and we enable a strict validation on the webhook

Reply:
Accepting only `on` and off` could surprise the user. Suppose you have set `hot_standby_feedback` to `yes` in your configuration, and it works because you don't have `pg_failover_slots` enabled.
You enable the pg_failover_slots extension, and the validation kicks in, rejecting a parameter that was valid before.

Reply:
I don't think this is an improvement. In the original, you can easily see the listing of "truthy" and "falsy" values.
With the map, you need to read the column on the right.
But, a `switch` statement would work well:
```suggestion
func ParsePostgresBoolean(in string) (bool, error) {
	sanitized := strings.TrimSpace(in)
	sanitized = strings.ToLower(sanitized)

	switch sanitized {
	case "1", "on", "yes", "true", "y", "t", "ye", "tr", "tru":
		return true, nil
	case "0", "no", "off", "false", "n", "f", "of", "fa", "fal", "fals":
		return false, nil
	default:
		return false, fmt.Errorf("configuration value is not a postgres boolean: %s", in)
	}
}
```